cmd/nerdctl/login.go

```diff
@@ -240,14 +241,31 @@ func loginClientSide(ctx context.Context, cmd *cobra.Command, auth types.AuthCon
 	if len(regHosts) == 0 {
 		return "", fmt.Errorf("got empty []docker.RegistryHost for %q", host)
 	}
+	needFallbackToPlainHttp := false
+	var needFallBackRegHosts []docker.RegistryHost
 	for i, rh := range regHosts {
 		err = tryLoginWithRegHost(ctx, rh)
 		identityToken := fetchedRefreshTokens[rh.Host] // can be empty
 		if err == nil {
 			return identityToken, nil
 		}
+		if insecure && (errutil.IsErrHTTPResponseToHTTPSClient(err) || errutil.IsErrConnectionRefused(err)) {
+			needFallbackToPlainHttp = true
+			needFallBackRegHosts = append(needFallBackRegHosts, rh)
+		}
```

Comment:
I reserve my opinion on https://github.com/containerd/nerdctl/pull/894#discussion_r827661658.

When it is not a valid HTTPs request and insecure-registry flaged, we can fallback directly to plain HTTP. cc @fahedouch

```suggestion
                err = tryLoginWithRegHost(ctx, rh)
		if err != nil && insecure && (errutil.IsErrHTTPResponseToHTTPSClient(err) || errutil.IsErrConnectionRefused(err)) {
		    rh.Scheme = "http"
		    err = tryLoginWithReghost(ctx, rh)
		}
		identityToken := fetchedRefreshTokens[rh.Host] // can be empty
		if err == nil {
			return identityToken, nil
		}
```

Reply:
OK, make sense